include/mppic/optimizers/xtensor/critics/local_goal_critic.hpp

```diff
@@ -0,0 +1,33 @@
+// Copyright 2022 FastSense, Samsung Research
+#pragma once
+
+#include "mppic/optimizers/xtensor/critic_function.hpp"
+#include "mppic/optimizers/xtensor/models/state.hpp"
+#include "mppic/utils.hpp"
+
+namespace mppi::xtensor::critics
+{
+
+class LocalGoalCritic : public CriticFunction
```

Comment:
Would it not be possible to have 2 instances of the goal critic and just have different settings for them for one to be 'local'?

Reply:
I'm not sure. GoalCritic penalize just based on last goal. It's simple and clear. 
LocalGoalCritic - calculate or get from ReferenceTrajectoryCritic furthest point that is nearest to one of the trajectories, apply offset on it, then take N points from offseted value, evaluate mean distance to them, if angle option turned on, then we offset furthest point on angle offset and use angle to this point as another critic. I added threshold to consider this critic (if we reach some % of full path by generated trajectories, it turned off)

Reply:
That sounds a little convoluted / heuristic crafted. What behavior is this providing that a reference trajectory critic could not?

Reply:
Reference trajectory critic not pushing you ahead of nearest goal. 
Idea was to use goal point that not actual last path point but some point that offset ahead of furthest nearest point